src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -349,14 +349,15 @@ private NuGetPackageManager GetNuGetPackageManager(string solutionDirectory)
             // and not have to determine if the package is a satellite package beforehand
             if (nuGetProjectContext.PackageExtractionContext == null)
             {
+                var revocationMode = SettingsUtility.GetRevocationMode(downloadContext.Settings);
```

Comment:
This is the first place where you've used `Settings` from downloadContext. In my opinion this should not be part of it, rather be part of `packageRestoreContext`.

Adding code randomly with different contexts make it really hard to maintain and understand for future.